executor/new_executor_xapi.go

```diff
@@ -51,17 +51,35 @@ type NewXExecutor interface {
 	AddLimit(l *plan.Limit) bool
 }
 
+// Closeable is a interface for closeable structures.
+type Closeable interface {
+	// Close closes the object.
+	Close() error
+}
+
+func closeAll(objs ...Closeable) error {
+	for _, obj := range objs {
+		if obj != nil {
+			err := obj.Close()
+			if err != nil {
+				return errors.Trace(err)
```

Comment:
I am not sure how to handle rest closeable objects. May be we could find a better way latter.